### PR TITLE
Added functional tests to test TVM compatibility for Linux images

### PR DESCRIPTION
**Pull request by vinaykotak**
Added functional tests to test if a Linux image is compatible for TrustedVM creation

### CODE REVIEW

File: XML/TestCases/FunctionalTests-TVM.xml [hunk 1]
```diff
@@ -10,8 +10,28 @@
...
+      <ARMImageName>Canonical 0003-com-ubuntu-server-trusted-vm 18_04-gen2 latest,RedHat RHEL 78-gen2 latest,RedHat RHEL 82gen2 latest,OpenLogic CentOS 7_8-gen2 latest,OpenLogic CentOS 8_2-gen2 latest,=~0003-com-ubuntu-server-trusted-vm\s18_04-gen2,=~RHEL\s78-gen2,=~RHEL\s82gen2,=~CentOS\s7_8-gen2,=~CentOS\s8_2-gen2</ARMImageName>
+    </SetupConfig>
+  </test>
+  <test>
+    <testName>TVM-TEST-COMPATIBILITY</testName>
+    <testScript>TVM-TestCompatibility.sh</testScript>
+    <files>.\Testscripts\Linux\TVM-TestCompatibility.sh,.\Testscripts\Linux\utils.sh,.\Tools\tvminfo</files>
+    <Platform>Azure</Platform>
+    <Category>Functional</Category>
+    <Area>TVM</Area>
```
Comment (harz566) [inline XML/TestCases/FunctionalTests-TVM.xml:24]: you may need to sync the master branch of LISAv2 and update the test case (TVM-TEST-SECUREBOOT-COMPATIBILITY, which has been updated from master branch) with the same Area and Tags

File: XML/TestCases/FunctionalTests-TVM.xml [hunk 2]
```diff
@@ -14,4 +14,23 @@
       <VMGeneration>2</VMGeneration>
     </SetupConfig>
   </test>
+  <test>
+    <testName>TVM-TEST-COMPATIBILITY</testName>
+    <testScript>TVM-TestCompatibility.sh</testScript>
+    <files>.\Testscripts\Linux\TVM-TestCompatibility.sh,.\Testscripts\Linux\utils.sh,.\Tools\tvminfo</files>
```
Comment (harz566) [inline XML/TestCases/FunctionalTests-TVM.xml]: Is it a tool that can be downloaded somewhere?

Comment (harz566) [reply XML/TestCases/FunctionalTests-TVM.xml]: TestHelpers.psm1 and .gitignore file should be updated accordingly after it has been published.

**Comment by juhlee-microsoft** [comment]
@vinaykotak Please post the change based test result here

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    exit 0
```
Comment (abhimarathe) [inline Testscripts/Linux/TVM-TestCompatibility.sh:8]: Please update state.txt before exit.

File: Testscripts/Linux/TVM-TestCompatibility.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    SetTestStateAborted
+    exit 0
+fi
+
+chmod +x "$homeDir"/tvminfo
+
+sudo "$homeDir"/tvminfo
+rc=$?
+
+if [ rc == 0]; then
```
Comment (abhimarathe) [inline Testscripts/Linux/TVM-TestCompatibility.sh]: if `rc` variable is only used once to store exit status, then you can use `$?` directly in if condition.
If there are plans to add code in between these two steps then it should be kept as it is.

**Comment by juhlee-microsoft** [comment]
@vinaykotak Are you still working on this PR? I don't see any update with the reviewers' comments.

**Comment by juhlee-microsoft** [comment]
> @vinaykotak Are you still working on this PR? I don't see any update with the reviewers' comments.

Found there is a backend work item should be clear before merging this PR. Hold until we got clear.

File: Testscripts/Linux/TVM-TestCompatibility.sh (hunk 3 above)
Comment (vinaykotak) [reply Testscripts/Linux/TVM-TestCompatibility.sh:8]: Done

File: XML/TestCases/FunctionalTests-TVM.xml (hunk 2 above)
Comment (vinaykotak) [reply XML/TestCases/FunctionalTests-TVM.xml]: Removed the tool from here. The test script is not downloading the tool from a public repo.

File: Testscripts/Linux/TVM-TestCompatibility.sh (hunk 4 above)
Comment (vinaykotak) [reply Testscripts/Linux/TVM-TestCompatibility.sh]: Will keep this as is for future use cases.

**Comment by LiliDeng** [comment]
@vinaykotak the new PR can replace this one? then we can close this PR safely?

**Comment by harz566** [comment]
@vinaykotak This PR  is not synced with master.

**Comment by harz566** [comment]
@vinaykotak  Sorry, feel free to reopen this PR, if it's still valid.

**Pull request closed without merging**